provisioning_templates/provision/kickstart_default.erb

```diff
@@ -94,14 +84,81 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 skipx
 
-<% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
-<% else -%>
-<% dhcp = !@static -%>
-<% end -%>
+<%
+  network_options = []
+  nameservers = []
+  subnet4 = iface.subnet
+  subnet6 = iface.subnet6
+
+  # device and hostname
+  network_options.push("--device=#{iface.mac || iface.identifier}")
+  network_options.push("--hostname #{@host.name}")
+
+  # single stack
+  if subnet4 && !subnet6
+    network_options.push("--noipv6")
+  elsif !subnet4 && subnet6
+    network_options.push("--noipv4")
+  end
+
+  # dual stack MTU check
+  raise("IPv4 and IPv6 subnets have different MTU") if subnet4 && subnet6 && subnet4.mtu.present? && subnet6.mtu.present? && subnet4.mtu != subnet6.mtu
+
+  # IPv4
+  if (subnet4 && !subnet4.dhcp_boot_mode?) || @static
+    network_options.push("--bootproto static")
+    network_options.push("--ip=#{iface.ip}")
+    network_options.push("--netmask=#{subnet4.mask}")
+    network_options.push("--gateway=#{subnet4.gateway}")
+  elsif subnet4 && subnet4.dhcp_boot_mode?
+    network_options.push("--bootproto dhcp")
+  end
+  if subnet4
+    nameservers.push(subnet4.dns_servers)
```

Comment:
This pushes an array onto an array. Not sure if this is allowed in safe mode, but this is probably what you intended.
```suggestion
    nameservers += subnet4.dns_servers
```

Reply:
It's quite the opposite, `push` is allowed but `+=` is not. That's why I don't use it: https://github.com/svenfuchs/safemode/blob/26d16d223630a0dbee3953aabbb1bfdd12da68ce/lib/safemode/core_jails.rb#L48-L55